Sources/MapboxCoreNavigation/CoreNavigationNavigator.swift

```diff
@@ -281,7 +281,6 @@ extension Navigator: ElectronicHorizonObserver {
 
 extension Navigator: NavigatorObserver {
     func onStatus(for origin: NavigationStatusOrigin, status: NavigationStatus) {
-        guard origin == .locationUpdate else { return }
```

Comment:
This change makes sense. That said, it might make this code redundant if the navigator doesn’t actually support synchronous leg changing anymore:

https://github.com/mapbox/mapbox-navigation-ios/blob/2543c009ea3e9675b1f1de78e51d8fb8b2d6a8da/Sources/MapboxCoreNavigation/RouteController.swift#L235-L237

Reply:
I've started a Q&A with NavNative team.